content/articles/2018-01-fs-1.0.0.Rmd

```diff
@@ -0,0 +1,176 @@
+
+**fs** functions always return 'tidy' paths. Tidy paths always expand `~`, use
+`/` to delimit directories and never have multiple `/` or trailing `/`.
+
+Tidy paths are also coloured (if your terminal supports it) based on the
+file permissions and file type. This colouring can be customised or extended by
+setting the `LS_COLORS` environment variable, in the same output format as [GNU
+dircolors](http://www.bigsoft.co.uk/blog/index.php/2008/04/11/configuring-ls_colors).
+
+![](/images/fs-1.0.0/colour.png)
+
+### Usage
+
+**fs** functions are divided into four main categories: manipulating paths
+(`path_`), files (`file_`), directories (`dir_`), and links (`link_`).
+Directories and links are special types of files, so `file_` functions
+will generally also work when applied to a directory or link.
+
+```{r}
+library(fs)
+
+# list files in the current directory
+dir_ls()
+
+# create a new directory
+tmp <- dir_create(file_temp())
+tmp
+
+# create new files in that directory
+file_create(path(tmp, "my-file.txt"))
+dir_ls(tmp)
+
+# remove files from the directory
+file_delete(path(tmp, "my-file.txt"))
+dir_ls(tmp)
+
+# remove the directory
+dir_delete(tmp)
+```
+
+**fs** is designed to work well with the pipe, although because it is a
+minimal-dependency infrastructure package it doesn't provide the pipe itself.
+You will need to attach [magrittr](http://magrittr.tidyverse.org) or similar.
+
+```{r}
+library(magrittr)
+
+paths <- file_temp() %>%
+  dir_create() %>%
+  path(letters[1:5]) %>%
+  file_create()
+paths
+
+paths %>% file_delete()
+```
+
+**fs** functions also work well in conjunction with other
+[tidyverse](https://www.tidyverse.org/) packages like
+[dplyr](http://dplyr.tidyverse.org) and [purrr](http://purrr.tidyverse.org).
+
+```{r}
+suppressMessages(
+  library(tidyverse))
+```
+
+Filter files by type, permission and size
+```{r}
+dir_info(recursive = TRUE) %>%
+  filter(type == "file", permissions == "u+r", size > "10KB") %>%
+  arrange(desc(size)) %>%
+  select(path, permissions, size, modification_time)
+```
+
+Display folder size
+```{r}
+dir_info(recursive = TRUE) %>%
+  group_by(directory = path_dir(path)) %>%
+  tally(wt = size, sort = TRUE)
+```
+
+Read a collection of files into one data frame. `dir_ls()` returns a named
+vector, so it can be used directly with `purrr::map_df(.id)`.
+```{r}
+# Create separate files for each species
+iris %>%
+  split(.$Species) %>%
+  map(select, -Species) %>%
+  iwalk(~ write_tsv(.x, paste0(.y, ".tsv")))
+
+# Show the files
+iris_files <- dir_ls(glob = "*.tsv")
+iris_files
+
+# Read the data into a single table, including the filenames.
+iris_files %>%
+  map_df(read_tsv, .id = "file", col_types = cols(), n_max = 2)
+
+file_delete(iris_files)
+```
+
+### Feedback wanted!
+
+We hope **fs** is a useful tool for both analysis scripts and packages.
+Please open [GitHub issues]() for any feature requests or bugs.
+
+In particular, we have found Windows users using non-ASCII filenames on
```

Comment:
I think there's a missing noun in this sentence somewhere— right now, it sounds like Windows users are tricky to reproduce (which could totally be true).